openedx/core/djangoapps/programs/tests/test_views.py

```diff
@@ -0,0 +1,135 @@
+"""
+Tests for the Programs.
+"""
+
+from mock import patch
+from provider.oauth2.models import Client
+from provider.constants import CONFIDENTIAL
+from unittest import skipUnless
+
+from django.conf import settings
+from django.test import TestCase
+
+from openedx.core.djangoapps.programs.views import get_course_programs_for_dashboard
+from openedx.core.djangoapps.programs.tests.mixins import ProgramsApiConfigMixin
+from student.tests.factories import AnonymousUserFactory, UserFactory
+
+
+@skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+class TestGetXSeriesPrograms(ProgramsApiConfigMixin, TestCase):
+    """
+    Tests for the Programs views.
+    """
+
+    def setUp(self, **kwargs):  # pylint: disable=unused-argument
+        super(TestGetXSeriesPrograms, self).setUp()
+        self.maxDiff = None
+        self.create_config(enabled=True, enable_student_dashboard=True)
+        Client.objects.get_or_create(name="programs", client_type=CONFIDENTIAL)
+        self.user = UserFactory()
+        self.programs_api_response = {
+            "results": [
+                {
+                    'category': 'xseries',
+                    'status': 'active',
+                    'subtitle': 'Dummy program for testing',
+                    'name': 'First Program',
+                    'course_codes': [
+                        {
+                            'organization': {'display_name': 'Test Organization', 'key': 'edX'},
+                            'display_name': 'Demo Course',
+                            'key': 'TEST_A',
+                            'marketing_slug': 'fake-marketing-slug',
+                            'run_modes': [
+                                {'sku': '', 'mode_slug': 'ABC_1', 'course_key': 'edX/DemoX_1/Run_1'},
+                                {'sku': '', 'mode_slug': 'ABC_2', 'course_key': 'edX/DemoX_2/Run_2'},
+                            ]
+                        }
+                    ]
+                }
+            ]
+        }
+
+    def test_get_course_programs_with_valid_user_and_courses(self):
```

Comment:
I have removed this test and created a new one with the name `test_get_course_programs_with_exception`